internal/cache/file/cache_handle.go

```diff
@@ -94,35 +91,43 @@ func (fch *CacheHandle) shouldReadFromCache(jobStatus *downloader.JobStatus, req
 }
 
 // checkEntryInFileInfoCache checks if entry is present for a given object in
-// file info cache with same generation. It returns nil if entry is present,
-// otherwise returns an error with appropriate message.
-func (fch *CacheHandle) checkEntryInFileInfoCache(bucket gcs.Bucket, object *gcs.MinObject, requiredOffset int64) error {
+// file info cache with same generation and at least requiredOffset.
+// It returns nil if entry is present, otherwise returns an appropriate error.
+// Whether to change the order in cache while lookup is controlled via
+// changeCacheOrder.
+func (fch *CacheHandle) checkEntryInFileInfoCache(bucket gcs.Bucket, object *gcs.MinObject, requiredOffset uint64, changeCacheOrder bool) error {
```

Comment:
Can we rename it to checkValidEntryInFileInfoCache, as we do some sort of validation on offset and generation.

Reply:
I think just "Entry" should be fine here because an entry is also valid if it has offset < requiredOffset. What the code is doing here is essentially checking whether the entry is present and if it is present whether it has offset and generation that is required by it. LMK if you think otherwise.

Reply:
Still feel like, `checkEntryInFileInfoCache` is too generic name. Can we rename it better if not checkValidEntryInFileInfoCache or checkRequiredEntryInFileInCache.